resources/views/layouts/default/flights/table.blade.php

```diff
@@ -69,23 +62,64 @@
         </div>
       </div>
       <div class="row">
-        <div class="col-sm-12 text-right">
-          @if ($simbrief !== false)
-            @if ($simbrief_bids === false || ($simbrief_bids === true && in_array($flight->id, $saved, true)))
-              <a href="{{ route('frontend.simbrief.generate') }}?flight_id={{ $flight->id }}"
-                 class="btn btn-sm btn-outline-primary">
-                Create SimBrief Flight Plan
-              </a>
+        <div class="col-sm-12 text-center">
+          <br>
+          <div class="row">
+            @if ($simbrief !== false)
+              @if ($simbrief_bids === false || ($simbrief_bids === true && in_array($flight->id, $saved, true)))
+                <div class="col-sm-6 text-left" style="vertical-align: bottom">
+                  <select id="aircraftselection" class="form-control select2" onchange="checkacselection()">
+                    <option value="ZZZZZ">Please Select An Aircraft</option>
+                    @if($flight->subfleets->isNotEmpty())
+                      @php $subfleets = $flight->subfleets; @endphp
+                    @else
+                      @php $userm = Auth::user(); $userSvc = app(App\Services\UserService::class); $subfleets = $userSvc->getAllowableSubfleets($userm); @endphp
```

Comment:
Which page is this showing up on? This needs to be moved to the controller also like I did before. Is it just the flight index?

Reply:
Oof this is within the context of a flight. I think this should be on the briefing page only, not here.

Reply:
This is the table.blade in flights , but we need to have a solution here too if we want to select the aircraft before reaching the simbrief_form.blade

And this the solution where va owner does not assign any fleets to his flights , then we need to get the list from the user

Reply:
Can this just be moved/handled only in the simbrief form? This is really heavy logic to put in here - for every flight, it's 1 call to get subfleets, then the fares for each subfleet, so a minimum of 2 calls - with 20 flights that's 40 separate calls, which aren't cached. And this also prevents the template from being cached.

Reply:
As I said , we need a solution to get the aircraft list before the simbrief_form.blade and this is displayed for bidded flights if the va owner does not assigns any subfleets to his flights ... 

If you move this to the simbrief_form , then we need to have a dropdown there to select the aircraft (which was my first solution and did not worked well to be honest)

If we can improve the flights controller and return the flights collection with the aircrafts according to what user can fly in the first place and handling situations where the va owner does not assign any subfleets to his flights then this will not be needed at all

Reply:
I can probably do that but I think moving it to the briefing page is the right place for it, not in this main flight listing. Can you remove this from here and move it to the simbrief form? The code should just be able to be moved. And then I will fix the controller.

Reply:
Result where admin assigns fleet/fleets to a flight (we get the flight->subfleets lists)

![ac1](https://user-images.githubusercontent.com/74361521/102832878-b5758a00-4400-11eb-8f2b-dee1a540d4cf.png)

Where not (we get users aircraft list)

![ac2](https://user-images.githubusercontent.com/74361521/102832950-d1792b80-4400-11eb-8443-7b006fa76d49.png)

If we can do this in the controller it would be great of course

Reply:
Yea in the default skin it would be too much there. So I think just move that to the simbrief form, as-is, and I'll fix in the controller. I moved other stuff into the controller so if you can pull this PR down and make sure it's all still working, that'd be great

Reply:
So you do not want to have aircraft selection in here at all , and all be done in simbrief_form ? did I understand correctly

Reply:
Correct, yes in the simbrief_form.

Reply:
Well , we are going back to where we started then ...

Anyway, i will look into it , if no aircraft_id is passed then I can render the selection dropdown and when user selects his aicraft and with a kind page reload he will see the final simbrief_form with all data mathching the flight and the selected aircraft.

Do not want to use javascript for aircraft selection (as the first thing I did in the beginning) , anywill work on it.

So are we closing this whole PR completely ?

Reply:
No, this PR is fine. What's wrong with moving the aircraft dropdown code that's in the flight selection into the simbrief form template? It should be able to just move over. I can look into doing it if you can't get it

Reply:
It was where we started, we needed the aicraft selection before all the calculations on simbrief_form.blade :) Anyway I need to work a little bit on it ... But I think I can handle it with a reload on the simbrief_form.blade upon aircraft selection if it fine for you too

Reply:
Ok don't do anything yet them - let me think of a way around this.

Reply:
If there was a selection page before the form, would that help? Would there be other things to add to that page, aside from selecting the aircraft, that would be helpful?

Reply:
The load calculations and the form fields required for simbrief plan generation relies on the aircraft selection (and of course subfleet it is a member of) 

But if a reload is ok, we can check it in the controller and if an aircraft is is not passed we can open the selection dropdown , upon selection page will reload itself with the id and goes on

Reply:
A reload is brittle. Lemme try adding an aircraft selection page before the simbrief page. I think that will be better, what do you think? So click "generate simbrief", a page to select the aircraft, and then the simbrief form? Lemme also look at which load calculations, maybe it's ok to do in JS on selection. But that is probably messy, unless it can load the extra data from the API. I do that on the PIREP form itself - if you look at the aircraft selection on the PIREP form, I load the fare information dynamically that way.

Reply:
I tried to this all in the simbrief_form with a dropdown and JS , it did not worked well ... Not impossible of course but I ended up implementing this solution 'cause did not like the final result and capabilities of enduser calculations. I wanted and needed all the calculations done on the server side to be honest.

A page before simbrief_form which passes the flight_id and the aircraft_id would work of course but I think we can handle it in the simbrief_form too , in the end user will see either 2 different pages or a page loaded second time. Not much a difference for the end user

Reply:
Currently we are checking the flight_id in the controler and I added the aircraft_id check to it too ... So if there is no aicraft_id passed I will open the dropdown , send the selection back to the page with &aircraft_id= as I did in the dropdown at table.blade 

If no aircraft_id is present , current controller was redirecting  the user to the bids page , I will remove that part and put a flash alert message and show the dropdown (rest of the page will not be rendered of course)

Reply:
How about a solution working like this ...

User reaches the simbrief_form without an aircraft selection
![sb1](https://user-images.githubusercontent.com/74361521/102839083-c4fccf00-4410-11eb-9f92-e3c38437d8fc.png)

Selects the aircraft here
![sb2](https://user-images.githubusercontent.com/74361521/102839094-c8905600-4410-11eb-943b-34d5073daffb.png)

Reaches the simbrief_form again with his flight and aircraft_id's returned
![sb3](https://user-images.githubusercontent.com/74361521/102839102-cc23dd00-4410-11eb-8105-c3333ee16d42.png)

With this, we can completely remove the aircraft selection dropdown from the flights.table_blade